core/templates/domain/statistics/learner-answer-details-backend-api.service.ts

```diff
@@ -16,52 +16,52 @@
  * @fileoverview Service to record learner answer info.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-require('domain/statistics/statistics-domain.constants.ajs.ts');
+import { StatisticsDomainConstants } from
+  'domain/statistics/statistics-domain.constants.ts';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
 
-angular.module('oppia').factory('LearnerAnswerDetailsBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService', 'SUBMIT_LEARNER_ANSWER_DETAILS_URL',
-  function(
-      $http, $q, UrlInterpolationService, SUBMIT_LEARNER_ANSWER_DETAILS_URL) {
-    var _recordLearnerAnswerDetails = function(
-        expId, stateName, interactionId, answer, answerDetails,
-        successCallback, errorCallback) {
-      var recordLearnerAnswerDetailsUrl = (
-        UrlInterpolationService.interpolateUrl(
-          SUBMIT_LEARNER_ANSWER_DETAILS_URL, {
+@Injectable({
+  providedIn: 'root'
+})
+export class LearnerAnswerDetailsBackendApiService {
+  constructor(private httpClient: HttpClient,
+    private urlInterpolationService: UrlInterpolationService) {}
+
+  recordLearnerAnswerDetails(
+      explorationId: string, stateName: string, interactionId: string,
+      answer: string, answerDetails: string): Promise<any> {
+    return new Promise((resolve, reject) => {
+      let recordLearnerAnswerDetailsUrl = (
+        this.urlInterpolationService.interpolateUrl(
+          StatisticsDomainConstants.SUBMIT_LEARNER_ANSWER_DETAILS_URL, {
             entity_type: 'exploration',
-            entity_id: expId
+            entity_id: explorationId
           }));
 
-      var payload = {
+      let payload = {
         state_name: stateName,
         interaction_id: interactionId,
         answer: answer,
         answer_details: answerDetails
       };
 
-      $http.put(recordLearnerAnswerDetailsUrl, payload).then(function(
-          response) {
-        if (successCallback) {
-          successCallback();
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
-    };
-
-    return {
-      recordLearnerAnswerDetails: function(expId, stateName, interactionId,
```

Comment:
Please do not change the file structure. Keep this in the same format. Define two functions:

1. `_recordLearnerAnswerDetails`: Mark this function as private
2. `recordLearnerAnswerDetails`: This will function 1.

Reply:
I combined them into one after reading the comments left by @brianrodri in #8831 .
What to do now? Should I revert it back to normal(2 functions)?

Reply:
I've left a comment here: https://github.com/oppia/oppia/pull/8831#discussion_r406734864. I'll wait for Brian's comment to conclude anything. 

Till the time I would say don't block yourself and go ahead and keep the same file structure (with two functions), since we've added along that lines previously. Though you might have to revert after my discussion with Brian.

Also, I guess this service won't take long now, could you also take up some other service from #8472 when you're done with this?

Reply:
@bansalnitish  I'll wait for your discussion. Meanwhile I'll pick a new Service.

Reply:
Hey @vinayakvvg, Please move ahead without changing the file structure i.e keep the functions as they were originally. Thanks and apologies for the one day delay on this.